src/service/VerifyService.ts

```diff
@@ -41,91 +40,100 @@ const defaultExpectedVersions: ExpectedVersions = {
     dockerCompose: '1.25.0',
 };
 
-export class VerifyService {
-    private readonly expectedVersions: ExpectedVersions;
-    public readonly semverOptions = { loose: true };
-
-    constructor(private readonly logger: Logger, expectedVersions: Partial<ExpectedVersions> = {}) {
-        this.expectedVersions = { ...defaultExpectedVersions, ...expectedVersions };
-    }
-
-    public async createReport(): Promise<VerifyReport> {
-        const lines: ReportLine[] = [];
-        const platform = `${os.type()} - ${os.release()} - ${os.platform()}`;
-        lines.push(await this.testNodeJs(this.expectedVersions.node));
-        const docker = await this.testDocker(this.expectedVersions.docker);
-        lines.push(docker);
-        lines.push(await this.testDockerCompose(this.expectedVersions.dockerCompose));
-        if (this.expectedVersions.symbolBootstrap) lines.push(await this.testSymbolBootstrap(this.expectedVersions.symbolBootstrap));
-        if (!docker.recommendation) lines.push(await this.testDockerRun());
-        if (!BootstrapUtils.isWindows()) {
-            lines.push(await this.testSudo());
-        }
-
-        return { lines, platform };
-    }
-
-    public logReport(report: VerifyReport): void {
-        this.logger.info(`OS: ${report.platform}`);
-        report.lines.forEach((line) => {
-            if (line.recommendation) {
-                this.logger.error(`${line.header}  - Error! - ${line.message} - ${line.recommendation}`);
-            } else {
-                this.logger.info(`${line.header} - OK! - ${line.message}`);
-            }
-        });
-    }
-
-    public validateReport(report: VerifyReport): void {
-        const errors = report.lines.filter((r) => r.recommendation);
-        if (errors.length) {
-            throw new Error(
-                'There has been an error. Check the report: \n' +
-                    errors.map((line) => ` - ${line.header}  - Error! - ${line.message} - ${line.recommendation}`).join('\n'),
-            );
-        }
-    }
+export interface VerifyAction {
+    shouldRun(lines: ReportLine[]): Promise<boolean>;
```

Comment:
why is this returning a promise?

Reply:
I was thinking that souldRun could be an async test. There is no case so far so back to plain boolean